Building Debian source package fails on PPA build environment

### Version of Apptainer

What version of Apptainer (or Singularity) are you using? Run

main and 1.1.5

### Expected behavior

What did you expect to see when you do...?

Successful Debian source package build.


### Actual behavior

What actually happened? Why was it incorrect?

I got following error.

```
make[2]: Leaving directory '/<<PKGBUILDDIR>>/debian/squashfuse-0.1.105'
failed to initialize build cache at /sbuild-nonexistent/.cache/go-build: mkdir /sbuild-nonexistent: permission denied
make[1]: *** [debian/rules:62: override_dh_auto_configure] Error 1
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
make: *** [debian/rules:58: build] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
```

default `GOCACHE` settings is incorrect on PPA build environment.

### Steps to reproduce this behavior

How can others reproduce this issue/problem?

Clone repository and build "Debian source package" `debuild -S --lintian-opts --display-info --show-overrides`

### What OS/distro are you running

```sh
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

### How did you install Apptainer

Write here how you installed Apptainer (or Singularity). Eg. RPM, source.

No installation. This is Apptainer Debian source package build error.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,13 +7,16 @@ For older changes see the [archived Singularity change log](https://github.com/a
 
 ## Changes Since Last Release
 
-### Changed defaults / behaviours
+### Bug fixes
 
 - Fix `GOCACHE` environment variable settings when building debian source
   package on PPA build environment.
 - Make `PS1` environment variable changeable via `%environment` section on
   definition file that used to be only changeable via `APPTAINERENV_PS1`
   outside of container. This makes container's prompt customizable.
+
+### Changed defaults / behaviours
+
 - When the kernel supports unprivileged overlay mounts in a user
   namespace, the container will be constructed using an overlay
   instead of underlay layout.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,8 @@ For older changes see the [archived Singularity change log](https://github.com/a
 
 ### Changed defaults / behaviours
 
+- Fix `GOCACHE` environment variable settings when building debian source
+  package on PPA build environment.
 - Make `PS1` environment variable changeable via `%environment` section on
   definition file that used to be only changeable via `APPTAINERENV_PS1`
   outside of container. This makes container's prompt customizable.

